### ISSUE TITLE
Substitutions at different levels doesn't always reach value

### ISSUE THREAD

**Issue by mjburton11** [hoburg/gpkit#1011]
In my aircraft creation model, I want to update one of the values in my vertical tail.  However, when I try substitutions like this: 
```python 
self.empennage.verticaltail.substitutions["V_v"] = 0.02
```
The  value doesn't update when I call `Aircraft()`.  But when I do the substitution like this: 
```python 
self.empennage.substitutions["V_v"] = 0.02
```
The value does update.  Is that supposed to happen?

**Comment by bqpd** [comment]
Only substitutions of top level constraintsets can be updated right now; my guess is that the only reason this works is that `empennage` has not yet been added to a larger constraintset.

**Comment by mjburton11** [comment]
That would be correct.  Well it seems to make sense to want to do substitutions at lower levels, right?

**Comment by bqpd** [comment]
We used to support that, but I changed my mind and now don't think it does; the eventual solve will happen with a single substitutions dictionary, which is the top level one. Each constraint set's substitutions get pulled into the new top level as during the creation of a higher-level constraint set containing them. This architecture is what makes it easy to free variables, and to see what values they will solve with, instead of having to search every submodel!

Right now the lower level constraintset's substitutions are mostly useful as (a) a historical artifact: what _were_ the constants of this submodel when it was created, and (b) if a single constraintset object is used in multiple models (which I've never seen done).
